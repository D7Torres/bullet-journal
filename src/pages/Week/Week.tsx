import { useParams } from 'react-router-dom'
import {
  getDatesOfWeek,
  toFriendlyShortWeekDate,
  toFriendlyWeekDate,
} from '@utils/dates'
import { useMemo } from 'react'
import { Day } from '@components/Day/Day'

export const Week = () => {
  const { year = new Date().getFullYear(), weekNumber = '1' } = useParams()

  const datesOfWeek = useMemo(
    () => getDatesOfWeek(+year, +weekNumber),
    [year, weekNumber],
  )

  if (!year || !weekNumber) {
    return null
  }

  return (
    <>
      <h1>
        {toFriendlyShortWeekDate(datesOfWeek[0])} -{' '}
        {toFriendlyShortWeekDate(datesOfWeek[6])}
      </h1>
      <ul>
        {datesOfWeek.map((date) => (
          <li key={date.getDay()}>
            <Day date={date} />
          </li>
        ))}
      </ul>
    </>
  )
}
