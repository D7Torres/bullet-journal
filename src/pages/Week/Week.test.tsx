import { render } from '@testing-library/react'
import { createMemoryRouter, RouterProvider } from 'react-router-dom'
import { Week } from './Week'
import { describe } from 'node:test'

const year = 2025
const weekNumber = 7
const memoryRouter = createMemoryRouter(
  [
    {
      path: `/journal/:year/week/:weekNumber`,
      element: <Week />,
    },
  ],
  {
    initialEntries: [`/journal/${year}/week/${weekNumber}`],
  },
)

describe('Journal page', () => {
  test('Journal page renders', () => {
    const { asFragment } = render(<RouterProvider router={memoryRouter} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
