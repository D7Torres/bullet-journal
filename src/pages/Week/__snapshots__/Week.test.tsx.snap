// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Journal page renders 1`] = `
<DocumentFragment>
  <h1>
    10 Feb - 16 Feb
  </h1>
  <ul>
    <li>
      <h2>
        Monday, 10
      </h2>
    </li>
    <li>
      <h2>
        Tuesday, 11
      </h2>
    </li>
    <li>
      <h2>
        Wednesday, 12
      </h2>
    </li>
    <li>
      <h2>
        Thursday, 13
      </h2>
    </li>
    <li>
      <h2>
        Friday, 14
      </h2>
    </li>
    <li>
      <h2>
        Saturday, 15
      </h2>
    </li>
    <li>
      <h2>
        Sunday, 16
      </h2>
    </li>
  </ul>
</DocumentFragment>
`;
